src/azul/plugins/metadata/hca/indexer/transform.py

```diff
@@ -1080,25 +1088,29 @@ def _sample_types(cls) -> FieldTypes:
             'effective_organ': null_str,
         }
 
-    class Sample:
+    class Sample[B: api.Biomaterial](metaclass=ABCMeta):
+        """
+        Neither this class nor any of its subclasses are ever instantiated. We
+        simply use it to associate custom functionality with types of
+        biomaterials that are used as samples.
+        """
         entity_type: str
-        api_class: type[api.Biomaterial]
+        api_class: type[B]
```

Comment:
In my previous review, I requested

```suggestion
        api_cls: type[B]
```